File: app/src/main/java/mishka/rssreader/data/DataManager.java
package mishka.rssreader.data;

import android.arch.lifecycle.LiveData;

import java.util.List;

import io.realm.RealmResults;
import mishka.rssreader.data.model.RealmRssItem;

public interface DataManager {
    void update();
    RealmResults<RealmRssItem> getRssItems();
    RealmRssItem getRssItemById(int id);
}

File: app/src/main/java/mishka/rssreader/data/DataManager.kt
package mishka.rssreader.data

import android.arch.lifecycle.LiveData

import mishka.rssreader.data.model.RssItem

interface DataManager {
    fun update()
    fun getPostById(postId: Int): LiveData<RssItem>
    fun getPosts(): LiveData<List<RssItem>>
}
